lib/matplotlib/path.py

```diff
@@ -206,12 +206,13 @@ def _fast_from_codes_and_verts(cls, verts, codes, internals=None):
         return pth
 
     def _update_values(self):
+        self._simplify_threshold = rcParams['path.simplify_threshold']
         self._should_simplify = (
             rcParams['path.simplify'] and
-            (len(self._vertices) >= 128 and
-             (self._codes is None or np.all(self._codes <= Path.LINETO)))
+            len(self._vertices) >= 128 and
+            (self._codes is None or np.all(self._codes <= Path.LINETO)) and
+            self._simplify_threshold > 0
```

Comment:
Perhaps that can go into a separate PR (linked to this discussion)?  If you look around mpl/__init__.py and mpl/rcsetup.py you should see how rcparam deprecation is handled.  That'll need a notice in api_changes too (this PR will probably benefit from an entry in the docs whatsnew too).